stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/extensions/ChatClient.kt

```diff
@@ -166,3 +169,25 @@ public fun ChatClient.loadOlderMessages(cid: String, messageLimit: Int): Call<Ch
         channelController.loadOlderMessages(messageLimit)
     }
 }
+
+/**
+ * Provides an instance of QueryChannelsRequest for next page.
+ *
+ * @param filter The filter for querying channels, see https://getstream.io/chat/docs/query_channels/?language=kotlin.
+ * @param sort The sort for the channels, by default will sort on last_message_at.
+ * @param limit The number of channels to retrieve.
+ * @param messageLimit How many messages to fetch per channel.
+ */
+@ExperimentalStreamChatApi
+public fun ChatClient.nextPageQueryChannelsRequest(
```

Comment:
Nit: This looks clumsy. I thought that when we release `OfflinePlugin` in VMs we can keep reference to `QueryChannelsState`. Then the next page request can be references much easier:
```kotlin
lateinit var state: QueryChannelsState

 fun initVM() {
   state = chatClient.asReferenced().queryChannels(request).asState(scope)
   ....
 }
 
 fun loadMore() {
  state.nextPagerRequest.value?.apply {limit = limit }?.let { chatClient.queryChannels(it) }?.enqueue()
 }

```

 But this extension is also ok to have. It's up to you to decide

Reply:
I was thinking of using the same thing but wasn't sure if this would be good. Thanks for the confirmation.